### PR TITLE
Added optional NodeRED component (Closes #146)

### LINKED ISSUE #146: Add NodeRED as optional component
> * This OH Node - https://www.npmjs.com/package/node-red-contrib-openhab22
> * BigTimer for setting schedules / sunrise-set events - https://flows.nodered.org/node/node-red-contrib-bigtimer
>  
> Those are the only 3rd party nodes I'm currently using. Another one, if they want more complete Astro replacement: https://flows.nodered.org/node/node-red-contrib-sunevents
> 
> Once key tutorial for more complex rules is around functions: http://nodered.org/docs/writing-functions.html

### PR DESCRIPTION
**Pull request by kubawolanin**
Here's my first contribution to openHABian. 
All suggestions welcome 😉 

Signed-off-by: Kuba Wolanin <hi@kubawolanin.com>

### CODE REVIEW

File: docs/openhabian.md [hunk 1]
```diff
@@ -239,6 +239,18 @@ Here are the passwords in question, their default value and the way to change th
...
+openHABian comes with a number of additional configs that allow you to quickly install home automation related software.
+
+* [frontail](https://github.com/mthenw/frontail) - openHAB Log Viewer accessible from [http://openHABianPi:9001/](http://openHABianPi:9001/)
+* [KNXd](http://michlstechblog.info/blog/raspberry-pi-eibknx-ip-gateway-and-router-with-knxd) - KNX daemon running at `224.0.23.12:3671/UDP`
+* [Homegear](https://www.homegear.eu/index.php/Main_Page) - Homematic control unit emulation
+* [Eclipse Mosquitto](http://mosquitto.org) - Open Source MQTT v3.1/v3.1.1 Broker
+* [OWServer](http://owfs.org/index.php?page=owserver_protocol) - 1wire control system
+* [Grafana](https://community.openhab.org/t/influxdb-grafana-persistence-and-graphing/13761/1) - persistence and graphing available from [http://openHABianPi:3000/](http://openHABianPi:3000/)
+* [Node-RED](https://nodered.org/) - Flow-based programming for the Internet of Things with preinstalled [openHAB2](https://flows.nodered.org/node/node-red-contrib-openhab2) and [BigTimer](https://flows.nodered.org/node/node-red-contrib-bigtimer) addons. Accessible from [http://openHABianPi:1880/](http://openHABianPi:1880/)
+
```
Comment (ThomDietrich) [inline docs/openhabian.md]: You probably wanted to remove this at the old location?
I'm not against moving these down here but they may get not as much attention down here, so better put a reference to here up there?

File: openhabian-setup.sh [hunk 2]
```diff
@@ -498,6 +498,29 @@ frontail() {
...
+  if [ $FAILED -eq 1 ]; then echo "FAILED (nodered)"; exit 1; fi
+  cond_redirect npm install -g node-red-contrib-bigtimer
+  if [ $? -ne 0 ]; then echo "FAILED (nodered bigtimer addon)"; exit 1; fi
+  cond_redirect npm update -g node-red-contrib-bigtimer
+  cond_redirect npm install -g node-red-contrib-openhab2
+  if [ $? -ne 0 ]; then echo "FAILED (nodered openhab2 addon)"; exit 1; fi
+  cond_redirect npm update -g node-red-contrib-openhab2
+  cond_redirect systemctl daemon-reload
+  cond_redirect /bin/systemctl enable nodered.service
+  cond_redirect /bin/systemctl restart nodered.service
```
Comment (ThomDietrich) [inline openhabian-setup.sh]: You can remove `/bin/` from here. I realize that I've added this by accident over in the frontail section. Please remove there as well ;)

File: openhabian-setup.sh [hunk 3]
```diff
@@ -1614,6 +1637,7 @@ show_main_menu() {
     "25 | Homegear"     "Homematic specific, the CCU2 emulation software Homegear" \
     "26 | knxd"         "KNX specific, the KNX router/gateway daemon knxd" \
     "27 | 1wire"        "1wire specific, owserver and related packages" \
+    "28 | NodeRED"      "Flow-based programming for the Internet of Things" \
```
Comment (ThomDietrich) [inline openhabian-setup.sh]: Come to think about it, let's move it up a bit, in front of the hardware-related components. I don't know why 24 is missing in this list but maybe that's a sign ;)

File: openhabian-setup.sh [hunk 4]
```diff
@@ -498,6 +498,29 @@ frontail() {
...
 }
 
+nodered() {
+  echo -n "$(timestamp) [openHABian] Installing Node-RED... "
+  if ! command -v npm &>/dev/null; then
+    cond_redirect wget -O - https://deb.nodesource.com/setup_7.x | bash -
+    if [ $? -ne 0 ]; then echo "FAILED (prerequisites)"; exit 1; fi
+    cond_redirect apt -y install nodejs
+    if [ $? -ne 0 ]; then echo "FAILED (nodejs)"; exit 1; fi
+  fi
```
Comment (ThomDietrich) [inline openhabian-setup.sh]: Second place where we need nodejs as an dependency. Let's move it to it's own function. wdyt?

**Comment by kubawolanin** [comment]
Hi @ThomDietrich thank you for reviewing my changes! I've pushed the fixes already. Hope it's all good now :)

Cheers!

**Pull request merged**